lib/main.dart

```diff
@@ -0,0 +1,271 @@
+import 'dart:convert';
+
+import 'package:bot_toast/bot_toast.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:http/http.dart' as http;
+import 'package:shimmer/shimmer.dart';
+import 'package:takehomeproject/ProductDetail.dart';
+import 'package:takehomeproject/ProductItem.dart';
+import 'package:cached_network_image/cached_network_image.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      debugShowCheckedModeBanner: false,
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      builder: BotToastInit(),
+      navigatorObservers: [BotToastNavigatorObserver()],
+      home: const MyHomePage(title: 'Take Home Project'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  List<ProductItem>? listProduct = [];
+  bool isFirstLoad = true;
+
+  Future fetchApi() async {
+    final url = Uri.parse('https://hf-android-app.s3-eu-west-1.amazonaws.com/android-test/recipes.json');
+    try {
+      final response = await http.get(url);
+      if(response.statusCode == 200) {
+        final extractedData = jsonDecode(response.body);
+        for (final data in extractedData) {
+          listProduct?.add(ProductItem.fromJson(data));
+        }
+      }
```

Comment:
No need to check status code. The Exception will thrown in catch blocked when the response is not success.

```suggestion
  Future<void> _fetchApi() async {
    try {
      final url = Uri.parse(
        'https://hf-android-app.s3-eu-west-1.amazonaws.com/android-test/recipes.json',
      );
      final response = await http.get(url);
      final extractedData = jsonDecode(response.body);

      if (extractedData is List) {
        for (final data in extractedData) {
          listProduct.add(ProductItem.fromJson(data));
        }
      } else {
        listProduct = [];
      }
    } catch (_) {
      BotToast.showSimpleNotification(
        title: 'Error fetching data',
        titleStyle: const TextStyle(color: CupertinoColors.white),
        duration: const Duration(seconds: 2),
        align: Alignment.bottomCenter,
        backgroundColor: Colors.red,
      );
      listProduct = [];
    }
    isFirstLoad = false;
    setState(() {});
  }
```